volumes.cpp

```diff
@@ -1301,18 +1321,156 @@ SurfaceVolume& makeSurface( const SurfaceCard* card, VolumeCache* v){
     else{
       throw std::runtime_error( mnemonic + " is not a supported surface" );
     }
-    
-    if( card->getTransform().hasData() ){
-      const Transform& transform = card->getTransform().getData();
-      surface->setTransform( &transform );
-    }
-    
-    cache.insert( card, surface );
-    return *surface;
-    
+  } 
+  if( card->getTransform().hasData() ){
+    const Transform& transform = card->getTransform().getData();
+    surface->setTransform( &transform );
   }
   
+  cache.insert( card, surface );
```

Comment:
I think this means that both facets and other surfaces are included in the cache, but I'm not sure the `card` used as the key for the cache is correct for facets.  This close brace moved in the `if` block for this, I think, and it makes sense because it is necessary to apply a transformation to both facets and to non-facet surfaces.